client/components/Profile.jsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Loader from "../components/Loader.jsx"
+
+const profileSubs = new SubsManager();
+
+const Profile = React.createClass({
+  mixins: [ReactMeteorData],
+
+  getInitialState: function() {
+    return {
+      nicknameChanged: false
+    };
+   },
+
+  getMeteorData () {
+    const usersHandle = profileSubs.subscribe("users");
+    let user = {};
+
+    if (usersHandle.ready()) {
+      user = Meteor.users.findOne(this.props.userProfile);
+    }
+    return {
+      usersHandle,
+      user
+    }
+  },
+
+  handleSubmit(e) {
+    e.preventDefault();
+    const nickname = ReactDOM.findDOMNode(this.refs.nickname).value.trim();
+    if (nickname.length === 0) return;
+    Meteor.call("user.changeNickname", nickname, (error, result) => {
+      if (!error) {
+        this.setState({nicknameChanged: true})
+      }
+    });
+  },
+
+  componentDidMount() {
+    if (this.data.usersHandle.ready() && this.props.userProfile == Meteor.userId()) {
```

Comment:
I ended up using the callback in the ref attribute which was easier and cleaner given that the form is conditonaly rendered. Removed reactive variables to meteor data